neps/nep-0001.md

```diff
@@ -100,6 +99,49 @@ the appropriate venue mentioned above. This gives the author a chance to flesh o
 initial concerns about the proposal.
 
+## The Proposal Process
+One of the primary goals of the NEP process is to prevent proposals being stuck in a limbo state of always being in progress, therefore NEP participants shall commit to always either accept or reject a `Draft` proposal.
+
+It can be rejected for the following reasons:
+
+* No-op proposal. It does not improve any of our values or velocity
+
+* Unjustifiable trade-offs. It makes the trade-offs in technological values or velocity that are not justifiable. Note, that if proposal is too difficult to implement or it adds too much architectural complexity it can be rejected as impeding future velocity too much.
+
+* Not ready. The proposal does not satisfy the required format, e.g. it does not explain all standard scenarios. Also, we don’t want the proposal discussion the be the place where the process is created, and so the authors and the proponents should come with a sufficiently ready proposal and do not expect to iterate on it with the community. They can however iterate on the proposal with the community in other open forums outside the official proposal process. Therefore, if proposal is submitted but community keeps finding gaps in the proposal, e.g. like reasonable alternatives, adversarial angles or gaps in the design, then even if author and proponents continue explaining them, after some time proposal runs out of time and is rejected.
+
+* Duplicate. A similar proposal has been already submitted and the author did not explain the difference sufficiently.
+
+* Spam. Someone is trying to get their proposal through by spamming;
+The time-line of the proposal is the following:
+
+### Initial phase (1 week at most)
+
+1.1. The process is initiated by an author submitting it in `Draft` status.
+
+1.2. The proposal is examined by the admins. If it deviates from the format the proposal's status is set to `Rejected` and allowed to be resubmitted in 1 month.
+
+1.3. Admins invite SMEs (some SMEs recognized by admins might join voluntarily) and representatives from the affected projects.
+
+1.4. (Optional) The stakeholders hire the auditors.
+
+1.5. Admins update the status to `Review` to indicate readiness for the next phase.
+
+### Discussion phase (2 months at most)
+
+2.1. SMEs and affected projects weigh-in.
+
+2.2. Auditors respond with their findings.
+
+2.3. Other community members join the discussion.
+
+2.4. Once the proposal is at the point where it is exhaustive, non-ambiguous, highly-detailed, and all findings are addressed and incorporated into the proposal, the admins make a decision on accepting or rejecting it.
+
+2.5. If proposal runs out of time it is rejected.
+
+### Implementation phase
+3.1. Admins invite representatives from the affected projects and the implementers to discuss the timeline of rolling out the feature. Once the timeline is agreed on, the remaining process is delegated to the implementers.
```

Comment:
I think this would work well for standards, but for protocol changes, at least right now we cannot afford to go through this very expensive process. I suggest that for protocol changes, the proposer also submit an accompanying prototype implementation to validate their proposal.

Reply:
@bowenwang1996 We could probably tighten these up if need be. For the sake of discussion, what do you think the protocol timelines would look like. If they are close I can adjust, if not, we might have to diverge the processes.

Reply:
@jlogelin I propose that we initially have the following process:
* Draft proposal with an accompanying prototype implementation to prove validity.
* A review period not exceeding two months where a committee would review the proposal and give feedback to iterate the proposal together with the author.
* Final decision made publicly during a public protocol meeting on whether to accept the proposal.